### PR TITLE
Updating windows service debug printing to use std::string

### PR DESCRIPTION
**Pull request by muffins**
This updates the `DebugPrintf` function to make use of `std::string` for ease of use in debugging services, as well as writes to `LOG(ERROR)`.

### CODE REVIEW

File: osquery/main/windows/main.cpp [hunk 1]
```diff
@@ -39,24 +39,14 @@ static SERVICE_STATUS kServiceStatus = {0};
...
-  }
+#define SLOG(s) ::osquery::DebugPrintf(s)
 
-  va_end(vl);
+void DebugPrintf(const std::string& s) {
+  auto dbgString = "[osqueryd] " + s;
+  if (isDebuggerPresent()) {
+    ::OutputDebugStringA(dbgString.c_str());
+  }
+  LOG(ERROR) << s;
```
Comment (muffins) [inline osquery/main/windows/main.cpp]: On second thought, not entirely sure this should be `LOG(ERROR)`, perhaps `LOG(INFO)` would be sufficient as we use this function for standard information messages also?

**Comment by osqueryer** [comment]
:-1: The commit bbb94402cebd68028adf6ea2fbbe1c597979abd9 (Job results: [2545](https://jenkins.osquery.io/job/osqueryPullRequestWindows/2545/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit bbb94402cebd68028adf6ea2fbbe1c597979abd9 (Job results: [2546](https://jenkins.osquery.io/job/osqueryPullRequestWindows/2546/)) failed one or more tests (**Windows**).

**Comment by muffins** [comment]
I've fixed this up a bit better in another diff, closing this out as it's not necessary.

**Pull request closed without merging**